rand_distr/src/binomial.rs

```diff
@@ -50,10 +46,34 @@ use rand::Rng;
 #[derive(Clone, Copy, Debug, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct Binomial {
-    /// Number of trials.
+    method: Method,
+    flipped: bool,
+    n: u64,
```

Comment:
Cool, I was not aware that Rust will put the `bool` into the padding of the tag :)

Reply:
Just to make sure we get the floatingpoint correct. How sure are you that Poisson is never used when flipped? It comes down to `1.0 - (1.0 - p) != 1.0` for any `p < 1.0`, it should be true, but I don't have a complete proof for it.

Reply:
Yes, Rust has niche-filling optimisations (even for other possible values of a `bool`; [example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6ff85a8c248970a1682fafa751242f63)).

The problem here is that the compiler won't look inside the `Binv` / `Btpe` structs right now. There are two ways we can work around this:

1. Use `#[repr(packed)]` on those structs, thus making them improperly aligned. Normally that's not a good solution (perf-wise), but since they are *only* ever used inside the `Method` enum, I *think* it's fine here.
2. We push the values (or just `flipped`) into `enum` fields: `enum Method { Binv(Binv, bool), ... }`. (There was interest in making `enum` variants types in their own right, e.g. RFC 2593, but that never (yet) happened.)

I suggest the latter, but either option should be fine here.